lib/scripting/post-process/class.ts

```diff
@@ -0,0 +1,218 @@
+	} else if (node.type === 'ConstructorMemberDeclaration') {
+		estree = ppConstructorMemberDeclaration(node);
+	} else if (node.type === 'RegularPropertyMemberDeclaration') {
+		estree = ppRegularPropertyMemberDeclaration(node);
+	} else if (node.type === 'PropertyGetDeclaration') {
+		estree = ppPropertyGetDeclaration(node);
+	} else if (node.type === 'PropertyLetDeclaration') {
+		estree = ppPropertyLetDeclaration(node);
+	} else if (node.type === 'PropertySetDeclaration') {
+		estree = ppPropertySetDeclaration(node);
+	}
+	return estree;
+}
+
+function ppClassDeclaration(node: ESIToken): any {
+	let id = identifier('undefined');
+	let constructor: MethodDefinition | undefined;
+	const methodDefinitions: MethodDefinition[] = [];
+	const varStmts: Statement[] = [];
+	const ids: string[] = [];
+	for (const child of node.children) {
+		if (child.type === 'Identifier') {
+			id = child.estree;
+		} else if (child.type === 'ClassMemberDeclaration') {
+			const memberDecl = child.children[0];
+			if (memberDecl.type === 'ConstructorMemberDeclaration') {
+				constructor = memberDecl.estree;
+			} else if (memberDecl.type === 'MethodMemberDeclaration') {
+				const functionDecl = memberDecl.estree as FunctionDeclaration;
+				const functionId = functionDecl.id as Identifier;
+				methodDefinitions.push(
+					methodDefinition(functionId, 'method', functionExpression(functionDecl.body, functionDecl.params)),
+				);
+				ids.push(functionId.name);
+			} else if (memberDecl.type === 'PropertyMemberDeclaration') {
+				methodDefinitions.push(memberDecl.estree);
+			} else if (
+				memberDecl.type === 'VariableMemberDeclaration' ||
+				memberDecl.type === 'ConstantMemberDeclaration'
+			) {
+				for (const varDecl of (memberDecl.estree as VariableDeclaration).declarations) {
+					const varId = varDecl.id as Identifier;
+					varStmts.push(
+						expressionStatement(
+							assignmentExpression(
+								memberExpression(thisExpression(), varId),
+								'=',
+								varDecl.init ? varDecl.init : identifier('undefined'),
+							),
+						),
+					);
+					ids.push(varId.name);
+				}
+			}
+		}
+	}
+	if (!constructor) {
+		constructor = methodDefinition(
+			identifier('constructor'),
+			'constructor',
+			functionExpression(blockStatement([]), []),
+		);
+	}
+	let body: ClassBody = classBody([constructor, ...methodDefinitions]);
+	body = replace(body, {
+		leave: (bodyNode, parentNode) => {
+			if (bodyNode.type === 'Identifier') {
+				if (parentNode !== null && parentNode.type !== 'MethodDefinition') {
+					if (ids.includes(bodyNode.name)) {
+						if (parentNode.type === 'MemberExpression') {
+							if (parentNode.object.type === 'Identifier') {
+								if (parentNode.object.name === bodyNode.name) {
+									return memberExpression(thisExpression(), bodyNode);
+								}
+							}
+						} else {
+							return memberExpression(thisExpression(), bodyNode);
+						}
+					}
+				}
+			}
+		},
+	}) as ClassBody;
```

Comment:
What does this do? Normalize case sensitivity?

Reply:
No. This is my poor attempt of your scope transformer. We cached every private/public var declaration in the class, and captured the id names. Then we walk the tree, looking for any identifiers, to add `this.` too.

I test this with the following:

`should transpile a class with identifiers that match member identifiers`

```
Class cvpmTimer
Public mBalls
Public Property Get Balls():Balls=mBalls.Keys:Test=x.mBalls:End Property
End Class
```

We need to make the first `mBalls` get's prefixed with `this`. The second mBalls is actually part a member of `x`. So that one we do not want to prefix with `this`.